.github/workflows/integration-test.yml

```diff
@@ -0,0 +1,59 @@
+name: Integration Test
+on: [push]
+
+jobs:
+  es-node-integration-test:
+    runs-on: self-hosted
+    timeout-minutes: 1440
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.21'
+
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '20.x'
+
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Deploy Contracts
+        run: |
+          cd ../storage-contracts-v1 
+          git checkout integration-test
+          git pull
+          npm install 
+          git submodule init && git submodule update
+          npx hardhat run scripts/deployL2-it.js --network qkc_testnet >> deploy.log
+          echo ES_NODE_CONTRACT_ADDRESS=`cat .caddr` >> "$GITHUB_ENV"
+          echo $ES_NODE_CONTRACT_ADDRESS
+
+      - name: Build and Run Bootnode Node
+        run: |
+          cd ../es-node
+          make
+          ./run-l2-it-rpc.sh > es-node-it-bootnode.log&
+
+      - name: Upload Blobs
+        run: |
+          cd ../ethstorage-sdk 
+          git checkout integration-test
+          git pull
+          npm install --force
+          rollup -c
+          node ituploader.js
+
+      - name: Test
+        run: |
+          cd ../es-node
+          env ES_NODE_STORAGE_MINER=0x5C935469C5592Aeeac3372e922d9bCEabDF8830d ES_NODE_SIGNER_PRIVATE_KEY=e690c916d02b6f3ad4bb70cc2663389ea88a79ded45613b1bd5545961d063258 ./run-l2-it.sh > es-node-it.log&
+          cd ./cmd/integration-test-server

```

Comment:
are there any success and failed tasks that we can inspect what they look like?

Reply:
Currently, we will check the final state after collected 2 hours node states, and failed service if any error collected.

```
	time.Sleep(executionTime)
	checkFinalState(lastRecord)

	if len(errorMessages) > 0 {
		panic(fmt.Sprintf("integration test fail %v", errorMessages))
	}
```